src/blocks/HighlightBlock/elements/Component.tsx

```diff
@@ -0,0 +1,135 @@
+'use client'
+import React from 'react'
+import { Carousel, CarouselContent, CarouselItem, type CarouselApi } from '@/components/ui/carousel'
+
+import { Card, CardContent, CardDescription, CardFooter } from '@/components/ui/card'
+import { H3 } from '@/components/ui/typography'
+import { Button } from '@/components/ui/button'
+import { Link } from '@payloadcms/ui'
+import { ArrowLeft, ArrowRight } from 'lucide-react'
+
+type Props = {
+  title: string
+  array: HighlightedElement[]
+  size: string
+  cardsToShow: string
+}
+
+/**
+type that contains the information needed to fill an element
+of the carousel
+*/
+export type HighlightedElement = {
+  name: string
+  summary?: string
+  image?: string
+  link: string
+}
+
+type SizeClasses = {
+  [key: string]: {
+    overallSize: string
+    header: string
+    card: string
+  }
+}
+
+const sizes: SizeClasses = {
+  medium: {
+    overallSize: 'w-1/2',
+    header: 'max-h-8',
+    card: 'h-48',
+  },
+  large: {
+    overallSize: 'w-1/2',
+    header: 'h-full',
+    card: 'min-h-72',
+  },
+  xl: {
+    overallSize: 'w-2/3',
+    header: 'h-full',
+    card: 'min-h-96',
+  },
+}
+
+export const HighlightElementsComponent: React.FC<Props> = ({
+  title,
+  array,
+  size,
+  cardsToShow,
+}) => {
+  const selectedSize = sizes[size]
+  const [api, setApi] = React.useState<CarouselApi>()
+  const [current, setCurrent] = React.useState(0)
+
+  React.useEffect(() => {
+    if (!api) {
+      return
+    }
+    setCurrent(api.selectedScrollSnap())
+    api.on('select', () => {
+      setCurrent(api.selectedScrollSnap())
+    })
+  }, [api])
```

Comment:
No need to handle state changes ourself if we use the  `<CarouselPrevious />` & `<CarouselNext />` components